### PR TITLE
Bookstore: - Feature/Redux

### PR DESCRIPTION
**Pull request by Bennyjoez**
## Bookstore: - Feature/Redux 🪶 

I managed to implement the following in line with the project requirements: 
1. ➕ Installed Redux toolkit and react redux. ✔️ 
2. ➕ Added a redux folder and added a store,js file ✔️ 
3. ➕ Added categories and book slices to manage the books' and categories' state ✔️ 
4. ➕ Added provider to index.js ✔️ 
5. ➕ Configure store.js. ✔️

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  initialState,
+  reducers: {
+    checkStatus: (state) => {
+      state.completed = state.categoriesArr.length > 0;
+    },
+  },
+});
+
+export const { checkStatus } = categoriesSlice.actions;
+export default categoriesSlice.reducer;
```
Comment (rotimiazeez) [inline src/redux/categories/categoriesSlice.js]: According to the project requirement, your categorySlice should include an array of categories (initial state: empty array) and a reducer that checks the status and always returns "Under construction" (the initial state should check to that string). This should look similar to the code below:
```
import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Bennyjoez 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good job integrating reducer and action into your app 👏🏽
- `@reduxjs/toolkit` & `react-redux` are in `package.json`. 😃
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes

_Check the comments under the review._

Please feel free to use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend considering them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (Bennyjoez) [reply src/redux/categories/categoriesSlice.js]: Hello @rotimiazeez 
Thank you for the suggestion provided, however, I was working with the project requirements, and they gave us a note saying **"NOTE: No changes in React components are required in this project."**
Also, the button is not in the wireframe provided for the project. The categories section only shows the text "Under construction". How then should I implement the button?

Comment (Bennyjoez) [reply src/redux/categories/categoriesSlice.js]: Changes to the categoriesSlice.js file have been implemented.

Comment (rotimiazeez) [reply src/redux/categories/categoriesSlice.js]: Hi @Bennyjoez,
Implementing the highlighted code here is fine

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Bennyjoez, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**